pkg/client/cli/intercept_state.go

```diff
@@ -29,33 +29,34 @@ import (
 	"github.com/telepresenceio/telepresence/v2/pkg/client/cli/extensions"
 )
 
-type interceptInfo struct {
-	sessionInfo
-
-	name        string
-	agentName   string
-	namespace   string
-	port        string
-	serviceName string
-	localOnly   bool
-
-	previewEnabled bool
-	previewSpec    manager.PreviewSpec
-
-	envFile string
-	envJSON string
-	mount   string // "true", "false", or desired mount point
-	toPod   []string
-
-	dockerRun   bool
-	dockerMount string // where to mount in a docker container. Defaults to mount unless mount is "true" or "false".
-
-	extState *extensions.ExtensionsState
-	extErr   error
+type interceptArgs struct {
+	name        string // Args[0] || `${Args[0]}-${--namespace}` // which depends on a combinationof --workload and --namespace
+	agentName   string // --workload || Args[0] // only valid if !localOnly
+	namespace   string // --namespace
+	port        string // --port // only valid if !localOnly
+	serviceName string // --service // only valid if !localOnly
+	localOnly   bool   // --local-only
+
+	previewEnabled bool                 // --preview-url // only valid if !localOnly
+	previewSpec    *manager.PreviewSpec // --preview-url-* // only valid if !localOnly
```

Comment:
This should probably not be a pointer given how it's used (causes the build failure).

Reply:
Yeah, I had to adjust it to get initialized... the linter was unhappy with it not being a pointer because I guess it contains a lock, so it didn't like being copied by value.